docker/lib/dependabot/docker/tag.rb

```diff
@@ -17,25 +21,30 @@ class Tag
           #{VERSION_WITH_SFX}|
           #{VERSION_WITH_PFX_AND_SFX}
       /x
-
+      sig { returns(String) }
       attr_reader :name
 
+      sig { params(name: String).void }
       def initialize(name)
         @name = name
       end
 
+      sig { returns(String) }
       def to_s
         name
       end
 
+      sig { returns(T::Boolean) }
       def digest?
         name.match?(FileParser::DIGEST)
       end
 
+      sig { returns(T::Boolean) }
       def looks_like_prerelease?
-        numeric_version.match?(/[a-zA-Z]/)
+        T.must(numeric_version).match?(/[a-zA-Z]/)
```

Comment:
Does this work?

```suggestion
        numeric_version&.match?(/[a-zA-Z]/)
```

Generally, we want to use Ruby's [safe navigation operator (`&.`)][1] where possible, and `T.must` only if safe navigation isn't possible. The reason being `T.must` can throw, when the parameter is `nil`.

[1]: https://thoughtbot.com/blog/ruby-safe-navigation